enumeration.go

```diff
@@ -302,6 +310,73 @@ func (e *Enumeration) Values() (interface{}, error) {
 	return strs, nil
 }
 
+// ExtendEnumeration extends an existing enumeration to add more values. The returned value should be
+// used with ArraySchemaEvolution.ApplyExtendedEnumeration to make changes persistent.
+func ExtendEnumeration[T EnumerationType](tdbCtx *Context, e *Enumeration, values []T) (*Enumeration, error) {
+	if len(values) == 0 {
+		return nil, fmt.Errorf("Error extending enumeration: empty values")
+	}
+
+	eName, err := e.Name()
+	if err != nil {
+		return nil, fmt.Errorf("Error extending enumeration: failed to get name of enumeration: %s", tdbCtx.LastError())
+	}
+
+	eType, err := e.Type()
+	if err != nil {
+		return nil, fmt.Errorf("Error extending enumeration: failed to get type of enumeration %s: %s", eName, tdbCtx.LastError())
+	}
+
+	tiledbType := enumerationTypeToTileDB[T]()
+	if eType != tiledbType {
+		return nil, fmt.Errorf("Error extending enumeration: type mismatch: enumeration type %v, values type %v", eType, tiledbType)
+	}
+
+	var cData unsafe.Pointer
+	var cDataLen C.uint64_t
+	var cOffsets unsafe.Pointer
+	var cOffsetsLen C.uint64_t
+
+	if tiledbType == TILEDB_STRING_ASCII {
+		var dataSize int
+		for _, v := range values {
+			dataSize += reflect.ValueOf(v).Len()
+		}
+		data := make([]byte, 0, dataSize)
+		offsets := make([]uint64, 0, len(values))
+		defer runtime.KeepAlive(data)
+		defer runtime.KeepAlive(offsets)
+		var currOffset uint64
+		for _, v := range values {
+			data = append(data, reflect.ValueOf(v).String()...)
```

Comment:
I wasn’t sure if `.String` here would return the stringy value itself or the result of the value’s `.String` method, so I tested it myself: https://go.dev/play/p/MBOtM1EnwZr

This will in fact do The Right Thing here, i.e. putting the bytes of the string into the slice rather than the tostringy string.

Maybe worth putting a custom type like this into the test cases to affirmatively document this (and verify it)? This could be as simple as replacing a `[]string` addition with a `[]bogus` type.

Reply:
@thetorpedodog Added test. Same code in in `newEnumeration`